### PR TITLE
[SALAD-23118] WebApp: Reusable modal component / GetNotifiedDemandChangesModal - added

### PR DESCRIPTION
**Pull request by vitto-moz**
image

### CODE REVIEW

**Comment by rners01** [comment]
waht about dropdown?
image

File: packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  },
+  singleValue: (baseStyles: CSSObjectWithLabel) => ({
+    ...baseStyles,
+    transition: 'opacity 300ms',
+    color: DefaultTheme.darkBlue,
+  }),
+}
+
+export const DropdownLight: FC<Omit<Props, 'customStyles' | 'classes'>> = ({ options, value, onChange }) => {
```
Comment (Maks19) [inline packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx:41]: Why do you need this component? You can just use one customStyles object

File: packages/web-app/src/modules/demand-alerts-views/constants.ts [hunk 2]
```diff
@@ -13,37 +9,3 @@ export const demandScenarios = [
   { label: 'Moderate Demand', value: 'moderateDemand' },
   { label: 'Low Demand', value: 'lowDemand' },
 ]
-
-export const customSetUpDropdownStyles: DropdownStylesConfig = {
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-alerts-views/constants.ts:17]: Why did you remove it?

File: packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  modalOverlay: {
+    position: 'fixed',
+    width: '100%',
+    height: '100%',
+    zIndex: 1000000000,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: '#00000070',
+    fontFamily: 'Mallory',
```
Comment (rners01) [inline packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx:20]: what is the proper approach, 
to use `theme.fontMallory` or existing one?

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx [hunk 4]
```diff
@@ -49,9 +52,41 @@ const styles: (theme: SaladTheme) => Record<string, CSS.Properties> = (theme: Sa
...
+  demandedHardwarePerformanceList,
+  classes,
+}) => {
+  const [isModalShown, setIsModalShown] = useState(true)
+
+  const updateTimerRef = useRef<NodeJS.Timeout | null>(null)
+
+  useEffect(() => {
+    fetchDemandedHardwarePerformanceList()
+    updateTimerRef.current = setInterval(fetchDemandedHardwarePerformanceList, oneHourInMilliseconds)
```
Comment (rners01) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx:73]: I haven't seen this in task description

File: packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx (hunk 1 above)
Comment (vitto-moz) [reply packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx:41]: This dropdown is used 3 times across the project and became kind of a standard, so I decided to combine the related functionality in one file and expose it like a component.

File: packages/web-app/src/modules/demand-alerts-views/constants.ts (hunk 2 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-alerts-views/constants.ts:17]: This styles are moved to DropdownLight component file (the exact and one place we we use it)

File: packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx (hunk 3 above)
Comment (vitto-moz) [reply packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx:20]: theme property is the correct one - I'll change

File: packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx (hunk 1 above)
Comment (Maks19) [reply packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx:41]: But it looks exactly like Dropdown component but with a different `customStyles` prop. Do you have some particular reason to make component and not use existing `customStyles` object

Comment (vitto-moz) [reply packages/web-app/src/components/Dropdown/DropdownLight/DropdownLight.tsx:41]: - reduce amount of imports
- expose functionality as a component for convenience
- use clear naming

File: packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx (hunk 3 above)
Comment (Maks19) [inline packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx:20]:
```suggestion
    fontFamily: theme.fontMallory,
```

File: packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    position: 'relative',
+    backgroundColor: theme.darkBlue,
+    overflow: 'hidden',
+    boxSizing: 'border-box',
+  },
+  closeIcon: {
+    position: 'absolute',
+    right: '16px',
+    top: '16px',
+    color: '#DBF1C1',
```
Comment (Maks19) [inline packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx:34]:
```suggestion
    color: theme.lightGreen,
```

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx [hunk 6]
```diff
@@ -60,11 +63,47 @@ const styles: (theme: SaladTheme) => Record<string, CSS.Properties> = (theme: Sa
...
+    }
+  }, [fetchDemandedHardwarePerformanceList])
+
+  const handleModalCloseClick = () => {
+    setIsModalShown(false)
+  }
+
+  const handleModalContinueClick = () => {
+    setIsModalShown(false)
+  }
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx:101]: Why do we have two functions that do the same?

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  saladImage: {
+    position: 'absolute',
+    top: '0px',
+    left: '0px',
+  },
+  closeIcon: {
+    position: 'absolute',
+    right: '16px',
+    top: '16px',
+    color: '#DBF1C1',
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx:62]:
```suggestion
    color: theme.lightGreen,
```

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+  return (
+    <ModalWithOverlay onCloseClick={onCloseClick}>
+      <div className={classes.modalContent}>
+        <h1 className={classes.title}>Get Notified Of Demand Changes</h1>
+        <p className={classes.description}>
+          You can set up alerts to be notified when the demand level of a GPU reaches your sweet spot, even at a
+          specific priority tier.
+        </p>
+        <div>
+          <h3 className={classes.subtitle}>Already have an account?</h3>
```
Comment (Maks19) [inline packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx:118]:
```suggestion
          <h2 className={classes.subtitle}>Already have an account?</h2>
```

File: packages/web-app/src/modules/reward-views/pages/RewardDetailsPage/ReviewAfterRedemption/ReviewAfterRedemption.tsx
Comment (Maks19) [inline packages/web-app/src/modules/reward-views/pages/RewardDetailsPage/ReviewAfterRedemption/ReviewAfterRedemption.tsx:1]: This section should be wider

image

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx (hunk 6 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/DemandMonitorPage.tsx:101]: this functional is a trigger for the second step
for now it just closes the modal

File: packages/web-app/src/modules/reward-views/pages/RewardDetailsPage/ReviewAfterRedemption/ReviewAfterRedemption.tsx
Comment (vitto-moz) [reply packages/web-app/src/modules/reward-views/pages/RewardDetailsPage/ReviewAfterRedemption/ReviewAfterRedemption.tsx:1]: fixed

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx (hunk 8 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx]: fixed

File: packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx (hunk 7 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/demand-monitor-views/DemandMonitorPage/GetNotifiedDemandChangesModal/GetNotifiedDemandChangesModal.tsx]: fixed

File: packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx (hunk 5 above)
Comment (vitto-moz) [reply packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx]: fixed

File: packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx (hunk 3 above)
Comment (vitto-moz) [reply packages/web-app/src/components/ModalWithOverlay/ModalWithOverlay.tsx:20]: fixed

**Review by Maks19 — APPROVED** [review]
